tree/tree/inc/ROOT/InternalTreeUtils.hxx

```diff
@@ -0,0 +1,57 @@
+/*************************************************************************
+ * Copyright (C) 1995-2021, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+/**
+ \file ROOT/InternalTreeUtils.hxx
+ \ingroup tree
+ \author Enric Tejedor Saavedra
+ \author Enrico Guiraud
+ \author Vincenzo Eduardo Padulano
+ \date 2021-03
+*/
+
+#ifndef ROOT_INTERNAL_TREEUTILS_H
+#define ROOT_INTERNAL_TREEUTILS_H
+
+#include <utility> // std::pair
+#include <vector>
+#include <string>
+
+class TTree;
+
+namespace ROOT {
+namespace Internal {
+/**
+\namespace ROOT::Internal::TreeUtils
+\ingroup tree
+\brief Namespace hosting functions and classes to retrieve tree information for internal use.
+*/
+namespace TreeUtils {
+
+using NameAlias = std::pair<std::string, std::string>; ///< A pair of name and alias of a TTree's friend tree.
+/**
+\struct ROOT::Internal::TreeUtils::FriendInfo
+\brief Information about friend trees of a certain TTree or TChain object.
+\ingroup tree
+*/
+struct FriendInfo {
```

Comment:
```suggestion
struct TFriendInfo {
```
Now that it is public and outside of class.

Reply:
Will be `RFriendInfo` in the next commit to comply with the new naming scheme